mitxgraders/comparers/comparers.py

```diff
@@ -112,30 +121,118 @@ class EqualityComparer(Comparer):
     True
     >>> grader(None, '[1/sqrt(2), 0, 1/sqrt(2)]')['ok']
     True
-    >>> FormulaGrader.reset_default_comparer()
+    >>> MatrixGrader.reset_default_comparer()
+
+    FormulaGrader and MatrixGrader both have set_default_comparer() and
+    reset_default_comparer() methods.
 
     """
     schema_config = Schema({
-        Required('transform', default=None): Nullable(is_callable)
+        Required('transform', default=None): All(
+            Nullable(is_callable),
+            # if f is None, coerce to identity function
+            lambda f: identity_transform if f is None else f
+        )
     })
 
-    def __call__(self, comparer_params_eval, student_eval, utils):
-        expected_input = comparer_params_eval[0]
-
+    @staticmethod
+    def validate(expected_eval, student_eval, utils):
         if hasattr(utils, 'validate_shape'):
             # in numpy, scalars have empty tuples as their shapes
-            shape = tuple() if isinstance(expected_input, Number) else expected_input.shape
+            shape = tuple() if isinstance(expected_eval, Number) else expected_eval.shape
             utils.validate_shape(student_eval, shape)
 
-        # If provided, apply the transform function
-        if self.config['transform'] is not None:
-            expected_input = self.config['transform'](expected_input)
-            student_eval = self.config['transform'](student_eval)
+    def __call__(self, comparer_params_eval, student_eval, utils):
+        expected_eval = comparer_params_eval[0]
+        self.validate(expected_eval, student_eval, utils)
 
-        return utils.within_tolerance(expected_input, student_eval)
+        transform = self.config['transform']
+        expected_eval = transform(expected_eval)
+        student_eval = transform(student_eval)
+
+        return utils.within_tolerance(expected_eval, student_eval)
 
 equality_comparer = EqualityComparer()
 
+class MatrixEntryComparer(CorrelatedComparer):
```

Comment:
I'm wondering if this should be `ArrayEntryComparer`, as it applies to vectors/matrices/tensors...

Reply:
Of all my comments, this one is probably the most work lol ;-)

Reply:
Actually, I've changed my mind on this one. We have `MatrixGrader`, so `MatrixEntryComparer` is fine. Maybe just add to the docstring that this does any array, not just matrices ;-)